pkg/ipmi/ipmi.go

```diff
@@ -173,3 +207,44 @@ func (i *IPMI) ShutoffWatchdog() error {
 
 	return nil
 }
+
+func (e *Event) Marshall() []byte {
+	buf := &bytes.Buffer{}
+	if err := binary.Write(buf, binary.LittleEndian, *e); err != nil {
+		log.Fatalf("Can't convert struct %v to bytes", *e)
+	}
+
+	return buf.Bytes()
+}
+
+func (i *IPMI) LogSystemEvent(sel []byte) error {
+	var data [16]byte
+
+	req := &req{}
+	req.msg.cmd = _BMC_ADD_SEL
+	req.msg.netfn = _IPMI_NETFN_STORAGE
+
+	//system event record
```

Comment:
How about moving all this logic to the marshal `method`?

Then you can do something like:

```
buf := event.Marshal()
req.msg.data = unsafe.Pointer(&buf[0])
```

Reply:
Updated it.